src/System.Management.Automation/engine/remoting/commands/CustomShellCommands.cs

```diff
@@ -964,7 +964,13 @@ private string ConstructPluginContent(out string srcConfigFilePath, out string d
                     }
                 }
 
-                string destPath = System.IO.Path.Combine(Utils.DefaultPowerShellAppBase, "SessionConfig",
+                string destFolder = System.IO.Path.Combine(Utils.DefaultPowerShellAppBase, "SessionConfig");
+                if (!Directory.Exists(destFolder))
+                {
+                    Directory.CreateDirectory(destFolder);
```

Comment:
Good catch! For example, on Linux, powershell is installed at /opt/microsoft/powershell, and you don't have permission to create a folder unless using `sudo`.